docs/src/main/asciidoc/mongodb-panache.adoc

```diff
@@ -528,7 +529,41 @@ you need to provide the value by yourself.
 ====
 
 `ObjectId` can be difficult to use if you want to expose its value in your REST service.
-So we created JSON-B and Jackson providers to serialize/deserialize them as a `String` which are are automatically registered if your project depends on one of the RESTEasy with JSON-B or RESTEasy with Jackson extensions.
+So we created JSON-B and Jackson providers to serialize/deserialize them as a `String` which are automatically registered if your project depends on one of the RESTEasy with JSON-B or RESTEasy with Jackson extensions.
+
+== Working with Kotlin Data classes
+
+Kotlin data classes are very convenient way to define data carrier classes, making them a great match to define an entity class.
+
+But this type of class comes with some limitations: all fields needs to be initialized at construction time or be marked at nullable,
+and the generated constructor needs to have as parameters all the fields of the data class.
+
+MongoDB with Panache uses the PojoCodec, this MongoDB codec mandates the presence of an empty (no parameter) constructor.
+
+Therefore, if you want to use a data class as an entity class, a way is needed to make Kotlin generate an empty constructor.
+To do so, you need to provide default values for all fields of your classes.
+The following sentence from the Kotlin documentation explain this:
+
+__On the JVM, if the generated class needs to have a parameterless constructor, default values for all properties have to be specified (see Constructors).__
+
+If for whatever reason the aforementioned solution is deemed unacceptable, there are alternatives
+
+Firstly, you can create a BSON Codec which will be automatically registered by Quarkus and will be used instead of the PojoCodec.
+See this part of the documentation: link:mongodb#simplifying-mongodb-client-usage-using-bson-codec[Using BSON codec]
+
+Secondly, you can use the `@BsonCreator` annotation to tell the PojoCodec to use the Kotlin data class default constructor,
+in this case all constructor parameters have  to be annotated with `@BsonProperty`: link:{mongodb-doc-root-url}/bson/pojos/#supporting-pojos-without-no-args-constructors[Supporting pojos without no args constructor]
+
+See an example of the Person class as a Kotlin data class:
```

Comment:
An example of a `Person` class defined as a Kotlin data class would look like:

Reply:
Thanks, it's easier to me :)